File: Exemplos M10/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/BRIDGE/Commands/IniciaVendaCredito.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService.BRIDGE.Commands

import com.elgin.intent_digital_hub.IntentDigitalHubService.BRIDGE.BridgeCommand

class IniciaVendaCredito(
    private val idTransacao: Int,
    private val pdv: String,
    private val valorTotal: String,
    private val tipoFinanciamento: Int,
    private val numeroParcelas: Int
) : BridgeCommand("IniciaVendaCredito") {
    override fun functionParameters(): String {
        return "\"idTransacao\"" + ":" + idTransacao + "," +
                "\"pdv\"" + ":" + "\"" + pdv + "\"" + "," +
                "\"valorTotal\"" + ":" + "\"" + valorTotal + "\"" + "," +
                "\"tipoFinanciamento\"" + ":" + tipoFinanciamento + "," +
                "\"numeroParcelas\"" + ":" + numeroParcelas
    }
}

File: Exemplos M10/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/IntentDigitalHubService/BRIDGE/inicia_venda_credito.dart
import 'package:intent_digital_hub/IntentDigitalHubService/BRIDGE/bridge_command.dart';

class IniciaVendaCredito extends BridgeCommand {
  final int _idTransacao;
  final String _pdv;
  final String _valorTotal;
  final int _tipoFinanciamento;
  final int _numeroParcelas;

  IniciaVendaCredito(this._idTransacao, this._pdv, this._valorTotal,
      this._tipoFinanciamento, this._numeroParcelas)
      : super('IniciaVendaCredito');

  @override
  functionParameters() {
    return '"idTransacao"'
        ':'
        '$_idTransacao'
        ','
        '"pdv"'
        ':'
        '"$_pdv"'
        ','
        '"valorTotal"'
        ':'
        '"$_valorTotal"'
        ','
        '"tipoFinanciamento"'
        ':'
        '$_tipoFinanciamento'
        ','
        '"numeroParcelas"'
        ':'
        '$_numeroParcelas';
  }
}
